MueLu: Add algorithm to enrich sparse graphs based on static powers

maxfirmbach opened the pull request:
@trilinos/muelu

## Motivation
<!--- 
Why is this change required?  What problem does it solve? Please link to a github 
issue that describes the problem/issue/bug this PR solves.
-->
Add a static pattern determination algorithm enriching the given pattern by applying powers of itself to it.
This should produce better quality sparse approximate inverses for ill-conditioned problems. Especially in the
case of block diagonal matrices (even with different block sizes) a really good approximation can be archived.
<!---
If applicable, let us know how this merge request is related to any other open
issues or pull requests:
-->
## Related Issues
* Related to  #10437, #10484, #10606 and #10909

## Stakeholder Feedback
<!--- 
If a github issue includes feedback from the relevant stakeholder(s), please link it.  
If the stakeholder(s) communicated that feedback through a different medium, please note that you did so.
-->
AMG for block matrices with schur complement

## Testing
<!---
Please confirm that any classes or functions in the Trilinos library that this PR touches are 
exercised by at least one test in Trilinos.  Please specify which test that is.  For untestable 
changes (e.g. changes to the nightly testing system) or changes to Trilinos tests, please say "N/A".
-->
- added unit tests for enriched sparse matrix graphs

<!--- 
## Additional Information
Anything else we need to know in evaluating this merge request?
 -->
Joined work with @mayrmt

cgcgcg commented on packages/muelu/src/Misc/MueLu_InverseApproximationFactory_def.hpp:
```diff
@@ -143,6 +160,60 @@ namespace MueLu {
+  InverseApproximationFactory<Scalar, LocalOrdinal, GlobalOrdinal, Node>::GetStaticPattern(const RCP<const CrsGraph>& sparsityPattern, int power) const {
+
+    // gather missing rows from other procs to generate an overlapping map
+    const RCP<const Import> rowImport = ImportFactory::Build(sparsityPattern->getRowMap(), sparsityPattern->getColMap());
```
Does sparsity pattern maybe already have an importer?

maxfirmbach replied:
I don't think so, at least I don't see were one has been initialized.

cgcgcg replied:
`sparsityPattern` is the graph of A, right? So you could check that `sparsityPattern->getImporter()` returns something non-null.

maxfirmbach replied:
Yep, I checked, it returns null :sweat_smile: .

mayrmt replied:
The `Xpetra::CrsGraph()` can return an importer via `getImporter()`. However, this is only implemented for the `Xpetra::TpetraCrsGraph`.

The `Xpetra::EpetraCrsGraph()` returns a hard-coded `Teuchos::null`. When looking at the implementation of the `Epetra_CrsGraph` in `Epetra`, it looks like the graph is capable of returning an importer. So, it might be just the incompleteness of `Xpetra` that comes into play here. (@maxfirmbach's, if you have tested this in your application code, then you have tested the `Epetra`- route through `Xpetra`.)

cgcgcg replied:
@mayrmt No, it's implemented for Epetra as well. The two specializations for EpetraNode return the importer.

cgcgcg commented:
It might be easier to perform matrix-matrix products to obtain the graph.

maxfirmbach commented:
> It might be easier to perform matrix-matrix products to obtain the graph.

I also thought about this in the beginning, but for performance reasons I don't need the explicitly calculated values. On the other hand the matrix-matrix product is so optimized it might be faster anyway. Not sure about this ...

cgcgcg commented:
Yes, you don't need the values. But I also think that you don't want to maintain an implementation of something that's similar to a matrix-matrix product. Also, I don't think that such a method belongs in MueLu. It would be nice if Tpetra could provide the graph of a matrix power.

maxfirmbach commented:
As our in-house code still uses Epetra I don't have access to Tpetra ... so an implementation in Xpetra would also be fine I guess.

trilinos-autotester commented:
Status Flag 'Pre-Test Inspection' - SUCCESS: The last commit to this Pull Request has been INSPECTED by label AT: PRE-TEST INSPECTED! Autotester is Removing Label; This inspection will remain valid until a new commit to source branch is performed.

trilinos-autotester commented:
Status Flag 'Pull Request AutoTester' - Testing Jenkins Projects:

<details>
<summary>
  <b>Pull Request Auto Testing STARTING</b> (click to expand)
</summary>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_PR_gcc-8.3.0</h3>
<ul>
  <li>Build Num: 988</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel7_sems-gnu-8.3.0-openmpi-1.10.1-openmp_release-debug_static_no-kokkos-arch_no-asan_no-complex_no-fpic_mpi_no-pt_no-rdc_no-uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: MueLu</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>10976</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>trilinos-any</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>MueLu-static-apriori-pattern-selection</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/maxfirmbach/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>f9841559ccc4bbb2bb74426bf32129018593f652</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>83e360dfdad394df2965fe10e29e960f2760513e</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_PR_gcc-7.2.0-serial</h3>
<ul>
  <li>Build Num: 988</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel7_sems-gnu-7.2.0-serial_release-debug_shared_no-kokkos-arch_no-asan_no-complex_no-fpic_no-mpi_no-pt_no-rdc_no-uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: MueLu</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>10976</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>trilinos-any</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>MueLu-static-apriori-pattern-selection</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/maxfirmbach/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>f9841559ccc4bbb2bb74426bf32129018593f652</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>83e360dfdad394df2965fe10e29e960f2760513e</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_PR_gcc-7.2.0-debug</h3>
<ul>
  <li>Build Num: 986</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel7_sems-gnu-7.2.0-openmpi-1.10.1-serial_debug_shared_no-kokkos-arch_no-asan_no-complex_no-fpic_mpi_no-pt_no-rdc_no-uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: MueLu</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>10976</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>trilinos-any</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>MueLu-static-apriori-pattern-selection</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/maxfirmbach/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>f9841559ccc4bbb2bb74426bf32129018593f652</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>83e360dfdad394df2965fe10e29e960f2760513e</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_PR_intel-17.0.1</h3>
<ul>
  <li>Build Num: 987</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel7_sems-intel-17.0.1-mpich-3.2-serial_release-debug_static_no-kokkos-arch_no-asan_no-complex_fpic_mpi_no-pt_no-rdc_no-uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: MueLu</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>10976</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>trilinos-any</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>MueLu-static-apriori-pattern-selection</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/maxfirmbach/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>f9841559ccc4bbb2bb74426bf32129018593f652</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>83e360dfdad394df2965fe10e29e960f2760513e</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_PR_clang-10.0.0</h3>
<ul>
  <li>Build Num: 1003</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel7_sems-clang-10.0.0-openmpi-1.10.1-serial_release-debug_shared_no-kokkos-arch_no-asan_no-complex_no-fpic_mpi_no-pt_no-rdc_no-uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: MueLu</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>10976</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>trilinos-any</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>MueLu-static-apriori-pattern-selection</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/maxfirmbach/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>f9841559ccc4bbb2bb74426bf32129018593f652</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>83e360dfdad394df2965fe10e29e960f2760513e</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_PR_cuda-10.1.243</h3>
<ul>
  <li>Build Num: 1003</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>ats2_cuda-10.1.243-gnu-8.3.1-spmpi-rolling_release_static_Volta70_Power9_no-asan_no-complex_no-fpic_mpi_pt_no-rdc_uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: MueLu</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>10976</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>vortex</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>MueLu-static-apriori-pattern-selection</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/maxfirmbach/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>f9841559ccc4bbb2bb74426bf32129018593f652</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>83e360dfdad394df2965fe10e29e960f2760513e</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_PR_python3</h3>
<ul>
  <li>Build Num: 816</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel7_sems-gnu-7.2.0-anaconda3-serial_debug_shared_no-kokkos-arch_no-asan_no-complex_no-fpic_no-mpi_no-pt_no-rdc_no-uvm_deprecated-on_pr-framework</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: MueLu</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>10976</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>ascic</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>MueLu-static-apriori-pattern-selection</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/maxfirmbach/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>f9841559ccc4bbb2bb74426bf32129018593f652</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>83e360dfdad394df2965fe10e29e960f2760513e</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_PR_cuda-11.4.2-uvm-off</h3>
<ul>
  <li>Build Num: 524</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel7_sems-cuda-11.4.2-sems-gnu-10.1.0-sems-openmpi-4.0.5_release_static_Volta70_no-asan_complex_no-fpic_mpi_pt_no-rdc_no-uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: MueLu</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>10976</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>GPU</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>MueLu-static-apriori-pattern-selection</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/maxfirmbach/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>f9841559ccc4bbb2bb74426bf32129018593f652</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>83e360dfdad394df2965fe10e29e960f2760513e</td>
  </tr>
</table>

<h2>Using Repos:</h2>

<dl>
  <dt>Repo: TRILINOS (maxfirmbach/Trilinos)</dt>
  <ul>
    <li>Branch: MueLu-static-apriori-pattern-selection</li>
    <li>SHA: f9841559ccc4bbb2bb74426bf32129018593f652</li>
    <li>Mode: TEST_REPO</li>
  </ul>
</dl>

Pull Request Author: maxfirmbach
</details>

jhux2 commented:
@maxfirmbach @mayrmt fyi, PR testing isn't working at the moment, see the pinned issue

trilinos-autotester commented:
Status Flag 'Pull Request AutoTester' - Jenkins Testing: 1 or more Jobs FAILED

Note: Testing will normally be attempted again in approx. 2 Hrs 30 Mins. If a change to the PR source branch occurs, the testing will be attempted again on next available autotester run.

<details>
<summary>
  <b>Pull Request Auto Testing has FAILED</b> (click to expand)
</summary>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_PR_gcc-8.3.0</h3>
<ul>
  <li>Build Num: 988</li>
  <li>Status: FAILED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel7_sems-gnu-8.3.0-openmpi-1.10.1-openmp_release-debug_static_no-kokkos-arch_no-asan_no-complex_no-fpic_mpi_no-pt_no-rdc_no-uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: MueLu</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>10976</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>trilinos-any</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>MueLu-static-apriori-pattern-selection</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/maxfirmbach/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>f9841559ccc4bbb2bb74426bf32129018593f652</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>83e360dfdad394df2965fe10e29e960f2760513e</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_PR_gcc-7.2.0-serial</h3>
<ul>
  <li>Build Num: 988</li>
  <li>Status: FAILED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel7_sems-gnu-7.2.0-serial_release-debug_shared_no-kokkos-arch_no-asan_no-complex_no-fpic_no-mpi_no-pt_no-rdc_no-uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: MueLu</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>10976</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>trilinos-any</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>MueLu-static-apriori-pattern-selection</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/maxfirmbach/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>f9841559ccc4bbb2bb74426bf32129018593f652</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>83e360dfdad394df2965fe10e29e960f2760513e</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_PR_gcc-7.2.0-debug</h3>
<ul>
  <li>Build Num: 986</li>
  <li>Status: FAILED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel7_sems-gnu-7.2.0-openmpi-1.10.1-serial_debug_shared_no-kokkos-arch_no-asan_no-complex_no-fpic_mpi_no-pt_no-rdc_no-uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: MueLu</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>10976</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>trilinos-any</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>MueLu-static-apriori-pattern-selection</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/maxfirmbach/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>f9841559ccc4bbb2bb74426bf32129018593f652</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>83e360dfdad394df2965fe10e29e960f2760513e</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_PR_intel-17.0.1</h3>
<ul>
  <li>Build Num: 987</li>
  <li>Status: FAILED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel7_sems-intel-17.0.1-mpich-3.2-serial_release-debug_static_no-kokkos-arch_no-asan_no-complex_fpic_mpi_no-pt_no-rdc_no-uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: MueLu</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>10976</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>trilinos-any</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>MueLu-static-apriori-pattern-selection</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/maxfirmbach/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>f9841559ccc4bbb2bb74426bf32129018593f652</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>83e360dfdad394df2965fe10e29e960f2760513e</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_PR_clang-10.0.0</h3>
<ul>
  <li>Build Num: 1003</li>
  <li>Status: FAILED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel7_sems-clang-10.0.0-openmpi-1.10.1-serial_release-debug_shared_no-kokkos-arch_no-asan_no-complex_no-fpic_mpi_no-pt_no-rdc_no-uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: MueLu</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>10976</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>trilinos-any</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>MueLu-static-apriori-pattern-selection</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/maxfirmbach/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>f9841559ccc4bbb2bb74426bf32129018593f652</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>83e360dfdad394df2965fe10e29e960f2760513e</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_PR_cuda-10.1.243</h3>
<ul>
  <li>Build Num: 1003</li>
  <li>Status: FAILED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>ats2_cuda-10.1.243-gnu-8.3.1-spmpi-rolling_release_static_Volta70_Power9_no-asan_no-complex_no-fpic_mpi_pt_no-rdc_uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: MueLu</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>10976</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>vortex</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>MueLu-static-apriori-pattern-selection</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/maxfirmbach/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>f9841559ccc4bbb2bb74426bf32129018593f652</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>83e360dfdad394df2965fe10e29e960f2760513e</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_PR_python3</h3>
<ul>
  <li>Build Num: 816</li>
  <li>Status: FAILED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel7_sems-gnu-7.2.0-anaconda3-serial_debug_shared_no-kokkos-arch_no-asan_no-complex_no-fpic_no-mpi_no-pt_no-rdc_no-uvm_deprecated-on_pr-framework</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: MueLu</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>10976</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>ascic</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>MueLu-static-apriori-pattern-selection</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/maxfirmbach/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>f9841559ccc4bbb2bb74426bf32129018593f652</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>83e360dfdad394df2965fe10e29e960f2760513e</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_PR_cuda-11.4.2-uvm-off</h3>
<ul>
  <li>Build Num: 524</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel7_sems-cuda-11.4.2-sems-gnu-10.1.0-sems-openmpi-4.0.5_release_static_Volta70_no-asan_complex_no-fpic_mpi_pt_no-rdc_no-uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: MueLu</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>10976</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>GPU</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_BRANCH</td>
    <td>MueLu-static-apriori-pattern-selection</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/maxfirmbach/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>f9841559ccc4bbb2bb74426bf32129018593f652</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>83e360dfdad394df2965fe10e29e960f2760513e</td>
  </tr>
</table>
</details>



<br>[CDash Test Results for PR# 10976.](https://trilinos-cdash.sandia.gov/index.php?project=Trilinos&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercount=2&showfilters=1&filtercombine=and&field1=buildname&compare1=63&value1=PR-10976-test&field2=buildstarttime&compare2=84&value2=NOW
)<br>

<br>[Wiki: How to Reproduce PR Testing Builds and Errors.](https://github.com/trilinos/Trilinos/wiki/Reproducing-PR-Testing-Errors)<br>

maxfirmbach commented:
This is my second take on this PR:

- moved the graph creation functionality out of `MueLu` to `Xpetra`
- simplified everything as proposed to now use a simple matrix-matrix product to indirectly get the graph of the result matrix, which should help with maintaining the method
-  also moved to a more simple loop instead of recursion

What do you think @cgcgcg  and @mayrmt ?

trilinos-autotester commented:
Status Flag 'Pre-Test Inspection' - - This Pull Request Requires Inspection... The code must be inspected by a member of the Team before Testing/Merging
           NO INSPECTION HAS BEEN PERFORMED ON THIS PULL REQUEST! - This PR must be inspected by setting label 'AT: PRE-TEST INSPECTED'.

cgcgcg commented on packages/xpetra/sup/Utils/Xpetra_GraphUtils.hpp:
```diff
@@ -0,0 +1,107 @@
+
+    @return Exponential CrsGraph with increased fill-in
+  */
+  static RCP<const CrsGraph> getExponentialCrsGraph(const RCP<const CrsGraph>& graph, int levelFill, Teuchos::FancyOStream &out) {
```
Is this correct? If I specify `levelFill=3` doesn't this give me `G^{4}`?

maxfirmbach replied:
Correct, in this version the levelFill is not representing the graph exponent directly, which yields the result you described. This could be changed of course … as right now the adjacency matrix is multiplied by itself.

maxfirmbach replied:
I changed the method to exactly produce what it states. So know levelFill=3 produces `G^{3}`, this should be more intuitive.

cgcgcg replied:
Cool, thank you!

trilinos-autotester commented:
Status Flag 'Pre-Test Inspection' - - This Pull Request Requires Inspection... The code must be inspected by a member of the Team before Testing/Merging
           ERROR: LABEL SET BY UNAPPROVED USER 'mayrmt', INSPECTION IS INVALID! - This PR must be re-inspected by approved user.

trilinos-autotester commented:
Status Flag 'Pre-Test Inspection' - - This Pull Request Requires Inspection... The code must be inspected by a member of the Team before Testing/Merging
           NO INSPECTION HAS BEEN PERFORMED ON THIS PULL REQUEST! - This PR must be inspected by setting label 'AT: PRE-TEST INSPECTED'.

trilinos-autotester commented:
Status Flag 'Pre-Test Inspection' - - This Pull Request Requires Inspection... The code must be inspected by a member of the Team before Testing/Merging
           ERROR: LABEL SET BY UNAPPROVED USER 'mayrmt', INSPECTION IS INVALID! - This PR must be re-inspected by approved user.

trilinos-autotester commented:
Status Flag 'Pre-Test Inspection' - - This Pull Request Requires Inspection... The code must be inspected by a member of the Team before Testing/Merging
           NO INSPECTION HAS BEEN PERFORMED ON THIS PULL REQUEST! - This PR must be inspected by setting label 'AT: PRE-TEST INSPECTED'.

trilinos-autotester commented:
Status Flag 'Pre-Test Inspection' - - This Pull Request Requires Inspection... The code must be inspected by a member of the Team before Testing/Merging
           ERROR: LABEL SET BY UNAPPROVED USER 'mayrmt', INSPECTION IS INVALID! - This PR must be re-inspected by approved user.

trilinos-autotester commented:
Status Flag 'Pre-Test Inspection' - SUCCESS: The last commit to this Pull Request has been INSPECTED by label AT: PRE-TEST INSPECTED! Autotester is Removing Label; this inspection will remain valid until a new commit to source branch is performed.

trilinos-autotester commented:
Status Flag 'Pull Request AutoTester' - Testing Jenkins Projects:

<details>
<summary>
  <b>Pull Request Auto Testing STARTING</b> (click to expand)
</summary>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_PR_gcc-8.3.0</h3>
<ul>
  <li>Build Num: 2776</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel7_sems-gnu-8.3.0-openmpi-1.10.1-openmp_release-debug_static_no-kokkos-arch_no-asan_no-complex_no-fpic_mpi_no-pt_no-rdc_no-uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: MueLu</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>10976</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>trilinos-any</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/maxfirmbach/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>55551c925b31cc8e6d187e47bbf408d036db91d5</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>73dcc21ae6078b22b4e281e7064bb2f221339cd0</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_PR_gcc-8.3.0-serial</h3>
<ul>
  <li>Build Num: 1275</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel7_sems-v2-gnu-8.3.0-serial_release-debug_shared_no-kokkos-arch_no-asan_no-complex_no-fpic_no-mpi_no-pt_no-rdc_no-uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: MueLu</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>10976</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>trilinos-any</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/maxfirmbach/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>55551c925b31cc8e6d187e47bbf408d036db91d5</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>73dcc21ae6078b22b4e281e7064bb2f221339cd0</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_PR_gcc-8.3.0-debug</h3>
<ul>
  <li>Build Num: 1265</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel7_sems-gnu-8.3.0-openmpi-1.10.1-serial_debug_shared_no-kokkos-arch_no-asan_no-complex_no-fpic_mpi_no-pt_no-rdc_no-uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: MueLu</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>10976</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>trilinos-any</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/maxfirmbach/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>55551c925b31cc8e6d187e47bbf408d036db91d5</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>73dcc21ae6078b22b4e281e7064bb2f221339cd0</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_PR_clang-11.0.1</h3>
<ul>
  <li>Build Num: 1264</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel7_sems-clang-11.0.1-openmpi-1.10.1-serial_release-debug_shared_no-kokkos-arch_no-asan_no-complex_no-fpic_mpi_no-pt_no-rdc_no-uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: MueLu</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>10976</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>trilinos-any</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/maxfirmbach/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>55551c925b31cc8e6d187e47bbf408d036db91d5</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>73dcc21ae6078b22b4e281e7064bb2f221339cd0</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_PR_python3</h3>
<ul>
  <li>Build Num: 2479</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel7_sems-gnu-7.2.0-anaconda3-serial_debug_shared_no-kokkos-arch_no-asan_no-complex_no-fpic_no-mpi_no-pt_no-rdc_no-uvm_deprecated-on_pr-framework</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: MueLu</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>10976</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>ascic</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/maxfirmbach/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>55551c925b31cc8e6d187e47bbf408d036db91d5</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>73dcc21ae6078b22b4e281e7064bb2f221339cd0</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_PR_cuda-11.4.2-uvm-off</h3>
<ul>
  <li>Build Num: 2267</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel7_sems-cuda-11.4.2-sems-gnu-10.1.0-sems-openmpi-4.0.5_release_static_Volta70_no-asan_complex_no-fpic_mpi_pt_no-rdc_no-uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: MueLu</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>10976</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>GPU</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/maxfirmbach/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>55551c925b31cc8e6d187e47bbf408d036db91d5</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>73dcc21ae6078b22b4e281e7064bb2f221339cd0</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_PR_intel-2021.3</h3>
<ul>
  <li>Build Num: 906</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel7_sems-intel-2021.3-sems-openmpi-4.0.5_release-debug_shared_no-kokkos-arch_no-asan_no-complex_fpic_mpi_no-pt_no-rdc_no-uvm_deprecated-off_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: MueLu</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>10976</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>trilinos-any</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/maxfirmbach/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>55551c925b31cc8e6d187e47bbf408d036db91d5</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>73dcc21ae6078b22b4e281e7064bb2f221339cd0</td>
  </tr>
</table>

<h2>Using Repos:</h2>

<dl>
  <dt>Repo: TRILINOS (maxfirmbach/Trilinos)</dt>
  <ul>
    <li>Branch: MueLu-static-apriori-pattern-selection</li>
    <li>SHA: 55551c925b31cc8e6d187e47bbf408d036db91d5</li>
    <li>Mode: TEST_REPO</li>
  </ul>
</dl>

Pull Request Author: maxfirmbach
</details>

trilinos-autotester commented:
Status Flag 'Pull Request AutoTester' - Jenkins Testing: 1 or more Jobs FAILED

Note: Testing will normally be attempted again in approx. 2 Hrs 30 Mins. If a change to the PR source branch occurs, the testing will be attempted again on next available autotester run.

<details>
<summary>
  <b>Pull Request Auto Testing has FAILED</b> (click to expand)
</summary>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_PR_gcc-8.3.0</h3>
<ul>
  <li>Build Num: 2776</li>
  <li>Status: FAILED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel7_sems-gnu-8.3.0-openmpi-1.10.1-openmp_release-debug_static_no-kokkos-arch_no-asan_no-complex_no-fpic_mpi_no-pt_no-rdc_no-uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: MueLu</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>10976</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>trilinos-any</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/maxfirmbach/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>55551c925b31cc8e6d187e47bbf408d036db91d5</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>73dcc21ae6078b22b4e281e7064bb2f221339cd0</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_PR_gcc-8.3.0-serial</h3>
<ul>
  <li>Build Num: 1275</li>
  <li>Status: FAILED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel7_sems-v2-gnu-8.3.0-serial_release-debug_shared_no-kokkos-arch_no-asan_no-complex_no-fpic_no-mpi_no-pt_no-rdc_no-uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: MueLu</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>10976</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>trilinos-any</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/maxfirmbach/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>55551c925b31cc8e6d187e47bbf408d036db91d5</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>73dcc21ae6078b22b4e281e7064bb2f221339cd0</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_PR_gcc-8.3.0-debug</h3>
<ul>
  <li>Build Num: 1265</li>
  <li>Status: FAILED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel7_sems-gnu-8.3.0-openmpi-1.10.1-serial_debug_shared_no-kokkos-arch_no-asan_no-complex_no-fpic_mpi_no-pt_no-rdc_no-uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: MueLu</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>10976</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>trilinos-any</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/maxfirmbach/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>55551c925b31cc8e6d187e47bbf408d036db91d5</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>73dcc21ae6078b22b4e281e7064bb2f221339cd0</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_PR_clang-11.0.1</h3>
<ul>
  <li>Build Num: 1264</li>
  <li>Status: FAILED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel7_sems-clang-11.0.1-openmpi-1.10.1-serial_release-debug_shared_no-kokkos-arch_no-asan_no-complex_no-fpic_mpi_no-pt_no-rdc_no-uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: MueLu</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>10976</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>trilinos-any</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/maxfirmbach/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>55551c925b31cc8e6d187e47bbf408d036db91d5</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>73dcc21ae6078b22b4e281e7064bb2f221339cd0</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_PR_python3</h3>
<ul>
  <li>Build Num: 2479</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel7_sems-gnu-7.2.0-anaconda3-serial_debug_shared_no-kokkos-arch_no-asan_no-complex_no-fpic_no-mpi_no-pt_no-rdc_no-uvm_deprecated-on_pr-framework</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: MueLu</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>10976</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>ascic</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/maxfirmbach/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>55551c925b31cc8e6d187e47bbf408d036db91d5</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>73dcc21ae6078b22b4e281e7064bb2f221339cd0</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_PR_cuda-11.4.2-uvm-off</h3>
<ul>
  <li>Build Num: 2267</li>
  <li>Status: FAILED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel7_sems-cuda-11.4.2-sems-gnu-10.1.0-sems-openmpi-4.0.5_release_static_Volta70_no-asan_complex_no-fpic_mpi_pt_no-rdc_no-uvm_deprecated-on_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: MueLu</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>10976</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>GPU</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/maxfirmbach/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>55551c925b31cc8e6d187e47bbf408d036db91d5</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>73dcc21ae6078b22b4e281e7064bb2f221339cd0</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: Trilinos_PR_intel-2021.3</h3>
<ul>
  <li>Build Num: 906</li>
  <li>Status: FAILED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>FORCE_CLEAN</td>
    <td>true</td>
  </tr>

  <tr>
    <td>GENCONFIG_BUILD_NAME</td>
    <td>rhel7_sems-intel-2021.3-sems-openmpi-4.0.5_release-debug_shared_no-kokkos-arch_no-asan_no-complex_fpic_mpi_no-pt_no-rdc_no-uvm_deprecated-off_no-package-enables</td>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>pkg: MueLu</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>10976</td>
  </tr>

  <tr>
    <td>PULLREQUEST_CDASH_TRACK</td>
    <td>Pull Request</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>TRILINOS</td>
  </tr>

  <tr>
    <td>TRILINOS_NODE_LABEL</td>
    <td>trilinos-any</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_REPO</td>
    <td>https://github.com/maxfirmbach/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_SOURCE_SHA</td>
    <td>55551c925b31cc8e6d187e47bbf408d036db91d5</td>
  </tr>

  <tr>
    <td>TRILINOS_SRN_CONFIG</td>
    <td>true</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_BRANCH</td>
    <td>develop</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_REPO</td>
    <td>https://github.com/trilinos/Trilinos</td>
  </tr>

  <tr>
    <td>TRILINOS_TARGET_SHA</td>
    <td>73dcc21ae6078b22b4e281e7064bb2f221339cd0</td>
  </tr>
</table>
</details>



<br>[CDash Test Results for PR# 10976.](https://trilinos-cdash.sandia.gov/index.php?project=Trilinos&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercount=2&showfilters=1&filtercombine=and&field1=buildname&compare1=63&value1=PR-10976-test&field2=buildstarttime&compare2=84&value2=NOW
)<br>

<br>[Wiki: How to Reproduce PR Testing Builds and Errors.](https://github.com/trilinos/Trilinos/wiki/Reproducing-PR-Testing-Errors)<br>

maxfirmbach commented:
Would someone mind posting the issue why the auto-tester failed? I assume it is related to the testing tolerances.

trilinos-autotester commented:
Status Flag 'Pre-Test Inspection' - - This Pull Request Requires Inspection... The code must be inspected by a member of the Team before Testing/Merging
           NO INSPECTION HAS BEEN PERFORMED ON THIS PULL REQUEST! - This PR must be inspected by setting label 'AT: PRE-TEST INSPECTED'.
